rafthttp/pipeline_test.go

```diff
@@ -227,5 +228,5 @@ func (t *roundTripperRecorder) Request() *http.Request {
 
 type nopReadCloser struct{}
 
-func (n *nopReadCloser) Read(p []byte) (int, error) { return 0, nil }
+func (n *nopReadCloser) Read(p []byte) (int, error) { return 0, io.EOF }
```

Comment:
read needs to specify the end of the data. We use ioutil.ReadAll on the body today, so it becomes a problem.